destination/writer/writer.go

```diff
@@ -126,53 +146,61 @@ func NewCSV(ctx context.Context, cfg *SnowflakeConfig) (*SnowflakeCSV, error) {
 	case compress.TypeCopy:
 		cmper = compress.Copy{}
 	default:
-		return nil, errors.Errorf("unrecognized compression type %q", cfg.Compression)
+		return nil, fmt.Errorf("unrecognized compression type %q", cfg.Compression)
+	}
+
+	tableFn, err := cfg.tableFunction()
+	if err != nil {
+		return nil, fmt.Errorf("failed to determine table function: %w", err)
 	}
 
 	return &SnowflakeCSV{
-		Prefix:            cfg.Prefix,
-		PrimaryKey:        cfg.PrimaryKey,
-		Stage:             cfg.Stage,
-		TableName:         cfg.TableName,
-		ProcessingWorkers: cfg.ProcessingWorkers,
-		FileThreads:       cfg.FileThreads,
+		prefix:            cfg.Prefix,
+		stage:             cfg.Stage,
+		processingWorkers: cfg.ProcessingWorkers,
+		fileThreads:       cfg.FileThreads,
 		db:                db,
-		evolver:           schema.NewEvolver(db),
 		compressor:        cmper,
-		insertsBuf:        &bytes.Buffer{},
-		updatesBuf:        &bytes.Buffer{},
-		compressedBuf:     &bytes.Buffer{},
+		buffers: sync.Pool{
+			New: func() any {
+				return bytes.NewBuffer(nil)
+			},
+		},
+		tableCache: make(map[string]snowflake.Table),
```

Comment:
Snowflake tables and their schema is now cached in the writer. We only retrieve it the first time. The assumption is that the table schema won't be modified externally, only through the connector.